File: environment/chapter1/mario.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
   void print(void);
   print();

}

void print(void)
{

    printf("Enter Height\n");
    int height = get_int("");
    if (height >= 24 || height < 0) //24 max height
    {
        printf("ERROR\n");
        print();
    }

    else
    {
        int heightPosition = height; //4 maxheight, prints top to bottom
        int reverseHeightPosition = height; //4
        int lengthAdder = 1;
        int reverseLengthAdder = 1;



        while (heightPosition > 0) //height: 4, while 4 > 0,heightPosition
        {
    for (int i = 0; i < height - lengthAdder; i++) //i = 0, 0 < 4 - 1 -> i < 3, print three empty spaces ---------------> i = 0, i < 4 - 2 -> i < 2, prints two spaces next heightPosition
            {
            printf(" ");//prints empty space
    }
    lengthAdder = lengthAdder + 1; //adds length to blocks, removes space printed
    for(int i = height; i > height - (lengthAdder - 1); i--) //i = 4, i > 4 - (2 - 1) ----> i = 4, i > 3, i--, print once, -------------> i = 4, i > 4 - (3 - 1), print two times
    {
        printf("#");

    }
    heightPosition = heightPosition - 1; //keeps track of height position, moves position down

    reverseLengthAdder = reverseLengthAdder - 1;
    printf("  ");//prints big gap

    //does reverse side
    for(int i = height; i < height - (reverseLengthAdder - 1); i++)
    {
        printf("#");
    }

   reverseHeightPosition = reverseHeightPosition + 1;
   printf("\n");
    }
    }

}


/*




/*/

File: environment/chapter6/sentimental/mario.py
import cs50
import math

while True:
    #
    print("Enter Height\n")
    height = cs50.get_int()
    if height >= 0 and height <= 23:
        # breaks free from the asking loop if right height, otherwise keeps asking
        break


for i in range(1, height + 1):
    # i adds each loop, height = 4 -> 1 - 5 -> spaces = 4 - 1 -> 3 spaces printed, blocks = 1, one block printed at end, no need to print spaces at other side, end="" controls newlines
    gapLength = 2
    blocks = i
    space = height - blocks

    print(" " * space, end="")
    print("#" * blocks, end="")

    print(" " * gapLength, end="")

    print("#" * blocks)